src/dataframerow/utils.jl

```diff
@@ -234,14 +260,20 @@ function group_rows(df::AbstractDataFrame, hash::Bool = true, sort::Bool = false
     if skipmissing
         popfirst!(starts)
         popfirst!(stops)
+        groups .-= 1
         ngroups -= 1
     end
 
     # sort groups if row_group_slots hasn't already done that
     if sort && !sorted
         group_perm = sortperm(view(df, rperm[starts], :))
+        group_invperm = invperm(group_perm)
```

Comment:
I'm very bad at permutations, but AFAICT that's the most efficient solution.

Reply:
Yeah, it's almost negligible compared with the time spent in grouping (this is only used for the slower fallback method).